### ISSUE TITLE
Question:Tuning SF static eval params based on Leela's  static evals

### ISSUE THREAD

**Issue by capilano** [official-stockfish/Stockfish#2452]
I was wondering if anyone has tried using neural network approaches to tune SF static eval params using Leela's static evals as ground truth. Specifically, my initial idea was to use high quality computer games/Leela self play games to generate test positions and filter out positions that lie outside the +- 5 range(actual SF eval) to remove outliers and then do the following.
1.) Gather all tunable (SF's) parameters and assign a range of values each one can take(maybe as a percentage of current value) to avoid non- sensical values and set each of them as learnable parameters. 
2.) Optimize SF evals based on Leela's static evals using some sort of softmax/cosine loss cost functions.
Assumptions: I am amusing that it is possible to obtain a fairly accurate eval conversion of SF eval into win/loss percentage.
Also, I have no idea about the underlying SF code or what the eval function even contains but I just was curious about whether there has been any work in this direction/whether it even makes sense from a theoretical perspective.Thanks

P.S: I initially wanted to use this approach to improve SF's pawn structure code. So, the idea was to try and come up with bonus terms for specific pawn structures and pieces that do well given that structure following a specific template for a large no of pre-coded pawn structures. For instance, something like the example below. 

1) Locked center-1: (Write down exact pawn structure including KS and QS pawns)
Common template for all pawn stuctures:
a) Are there knights left/How many: Yes/No, 0/1/2
b) if a is true, potential access squares(f6,e5,d5,c3,f3 etc etc)
c) Same logic as above but for enemy knights
Now assign bonus/penalty terms/ Try to capture as many possible terms based on no of possible squares and any other miscellaneous factors.(All of them will be tunable/learnable parameters).
d) Proceed to bishops(Add as many heuristics as you need as long as it remotely makes sense) and add bonus/penalty terms, similar to the knight example
e)Do the same for other pieces
f) Use king safety, space etc as modulating terms for the above heurestics and assign learnable bonuses.
g...z) Complete this common template so that it captures a wide variety of positional considerations needed to evaluate pawn structures/what pieces do well and assign learnable parameters. 
2) Locked center-2( Write down exact pawn structure ......)
3) Locked center-3(...)
4) Open center-1...n(..)
5) Two pawn islands
6)Go on for as many pawn structures that make sense and use the same template and figure out bonus/penalty terms. 
7) Unknown pawn structure-Special case. use the same penalty/bonus terms as pawn structure that is closest to it and that has already been coded in.
8) Optimize for all tunable params and get rid of terms from the template that make little  difference to eval.(Do this for each pawn structure separately)

**Comment by Sopel97** [comment]
Around a year ago I tries to tune around 1000 stockfish parameters for static eval with a gradient descent optimizer based on leela's evaluations. The results were a disaster.

**Comment by capilano** [comment]
Ok. Thanks. How many examples did you use? If the error did not go down, then i guess there is not much else left to do because it's likely that way more parameters are needed and in this case, i guess it's not practical because someone would have to handcraft them.

**Comment by Sopel97** [comment]
I don't remember exactly and I don't have the data anymore, but somewhere around a million positions and I tried a few variations of the learning program.
